cl_sii/rut/__init__.py

```diff
@@ -144,7 +145,10 @@ def __hash__(self) -> int:
     def clean_str(cls, value: str) -> str:
         # note: unfortunately `value.strip('.')` does not remove all the occurrences of '.' in
         #   'value' (only the leading and trailing ones).
-        return value.strip().lstrip('0').replace('.', '').upper()
+        cleaned_value = value.strip().replace('.', '').upper()
+        # Remove leading zeros excepts it is followed by a '-', so we can accept the RUT '0-0'
+        leading_zero_free_value = re.sub(r'^(0+)(?!-)', '', cleaned_value)
```

Comment:
This regular expression does preserve the previous behavior of `Rut.clean_str('000')` returning an empty string, but given that the spirit of this pull request is avoiding the complete disappearance of zeroes, we could improve the code in the following way:

```suggestion
        leading_zero_free_value = re.sub(r'^0+(\d+)', r'\1', cleaned_value)
```

With the above change, the return value of `Rut.clean_str('000')` would be `'0'`, which even though is not a valid RUT, it makes more sense than an empty string.

Reply:
Great catch, thanks!